riscv/src/compiler.rs

```diff
@@ -848,14 +889,53 @@ fn only_if_no_write_to_zero_vec(statements: Vec<String>, reg: Register) -> Vec<S
     }
 }
 
-static COPROCESSOR_SUBSTITUTIONS: &[(&str, &str)] =
-    &[("poseidon_coprocessor", "x10 <== poseidon(x10, x11);")];
+fn poseidon_gl_call() -> String {
+    let decoding = |i| {
+        format!(
+            r#"
+        addr <=X= {} + x10;
+        P{i} <== mload();
+        addr <=X= {} + x10;
+        tmp1 <== mload();
+        P{i} <=X= P{i} + tmp1 * 2**32;
```

Comment:
Actually, that's not what I mean. I though that would already be enforced somehow (because memory words are always 32 bits?).

What I mean is that `P{i} + tmp1 * 2**32` < p. Not sure what the overhead of this is, but we could just add a `assert!(x < P_GL);` for each `u64` input `x` in rust code?

Reply:
ah yes, I wanted to do that but forgot